File: python/count-asterisks.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        bar_open = False
        asterisk_count = 0
        for c in s:
            if c == '|':
                bar_open ^= True
            if c == '*' and not bar_open:
                asterisk_count += 1
        return asterisk_count

File: rust/count-asterisks.rs
impl Solution {
    pub fn count_asterisks(s: String) -> i32 {
        let mut bar_open = false;
        let mut asterisk_count = 0;
        for c in s.chars() {
            if c == '|' {
                bar_open = bar_open ^ true;
            }
            if c == '*' && !bar_open {
                asterisk_count += 1;
            }
        }
        return asterisk_count;
    }
}
